File: Baekjoon/Class1/36.10809.py
import sys
input = sys.stdin.readline

FIRST_ORD = ord("a")

s = input().strip()
first_appearance = [-1] * 26
for i, ch in enumerate(s):
    index = ord(ch) - FIRST_ORD
    if first_appearance[index] == -1:
        first_appearance[index] = i
print(" ".join(map(str, first_appearance)))
#print(*(first_appearance))

File: Baekjoon/Class1/36.10809.js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim();

const s = input;
const firstAppearance = new Array(26).fill(-1);
for(let i = 0; i < s.length; i++) {
    const index = s.charCodeAt(i) - 97; // 'a' = 97
    if(firstAppearance[index] == -1)
        firstAppearance[index] = i;
}
console.log(firstAppearance.join(" "));